internal/services/applicationinsights/application_insights_api_key_resource.go

```diff
@@ -160,7 +160,7 @@ func resourceApplicationInsightsAPIKeyRead(d *pluginsdk.ResourceData, meta inter
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := apikeys.ParseApiKeyID(d.Id())
+	id, err := apikeys.ParseApiKeyIDInsensitively(d.Id())
```

Comment:
This means we'll have Resources using the incorrect Resource IDs in the state - can we revert this and instead add a state migration to fix the Resource ID?